File: pep1-frontend/pep1/src/services/bonusbrand.service.js
import axios from 'axios';

const API_URL = 'http://localhost:8090/api/v1/bonusBrand';

function getBonusBrandById(id) {
    return axios.get(`${API_URL}/${id}`);
}

function saveBonusBrand(bonusBrand) {
    return axios.post(`${API_URL}/`, bonusBrand);
}

function deleteBonusBrand(id) {
    return axios.delete(`${API_URL}/${id}`);
}

function updateBonusBrand(bonusBrand) {
    return axios.put(`${API_URL}/`, bonusBrand);
}

function getBonusByBrand(brand) {
    return axios.get(`${API_URL}/bonus/${brand}`);
}
function getAllBonusBrands() {
    return axios.get(`${API_URL}/`);
}

export default {
    getBonusBrandById,
    getAllBonusBrands,
    saveBonusBrand,
    deleteBonusBrand,
    updateBonusBrand,
    getBonusByBrand
}

File: pep1-backend/src/main/java/usach/tingeso/services/BonusBrandService.java
package usach.tingeso.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import usach.tingeso.entities.BonusBrandEntity;
import usach.tingeso.repositories.BonusBrandRepository;

import java.util.List;

@Service
public class BonusBrandService {
    @Autowired
    BonusBrandRepository bonusBrandRepository;

    // Basic CRUD operations
    public BonusBrandEntity getBonusBrandById(Long id){
        if (bonusBrandRepository.findBonusBrandByIdBonus(id) == null)
            return null;
        return bonusBrandRepository.findBonusBrandByIdBonus(id);
    }
    public List<BonusBrandEntity> getAllBonusBrands(){
        List<BonusBrandEntity> bonusBrands = bonusBrandRepository.findAll();
        if (bonusBrands.isEmpty())
            return null;
        return bonusBrands;
    }
    public BonusBrandEntity saveBonusBrand(BonusBrandEntity bonusBrand){
        if (bonusBrand == null)
            return null;
        return bonusBrandRepository.save(bonusBrand);
    }
    public void deleteBonusBrand(Long id){
        if (bonusBrandRepository.findBonusBrandByIdBonus(id) == null)
            return;
        bonusBrandRepository.deleteById(id);
    }
    public BonusBrandEntity updateBonusBrand(BonusBrandEntity bonusBrand){
        if (bonusBrand == null)
            return null;
        return bonusBrandRepository.save(bonusBrand);
    }

    // Get the bonus by brand
    public int getBonusByBrand(String brand){
        if (brand == null || brand.isEmpty())
            return 0;
        List<BonusBrandEntity> bonusBrandEntities = bonusBrandRepository.findHighestActiveBonusByBrand(brand);
        if (bonusBrandEntities.isEmpty())
            return 0;
        return bonusBrandEntities.get(0).getBonus();
    }

    // EOF
}
